client/components/FormattingBar/controlComponents/ControlsReference.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState, useMemo, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { Button } from '@blueprintjs/core';
+
+import { MenuButton, MenuItem } from 'components/Menu';
+import { getReferenceableNodes } from 'components/Editor/utils/references';
+
+const propTypes = {
```

Comment:
No time like the present to start making full use of typescript, if you've got the time to convert this (now or in another PR)